packages/flutter/lib/src/services/binding.dart

```diff
@@ -29,6 +32,20 @@ mixin ServicesBinding on BindingBase {
   static ServicesBinding get instance => _instance;
   static ServicesBinding _instance;
 
+  /// The default instance of [BinaryMessenger].
+  ///
+  /// This is used to send messages from the application to the platform, and
+  /// keeps track of which handlers have been registered on each channel so
+  /// it may dispatch incoming messages to the registered handler.
+  BinaryMessenger defaultBinaryMessenger;
```

Comment:
This is public and mutable.  We should probably make it private with a public getter.

Reply:
Hmm... seems that won't work, since we'll need to set it in subclasses. 
Thus, the setter has to be visible in its subclasses, otherwise we don't have a way to modify it in a subclass.

Reply:
Then what about changing the  signature of `initBinaryMessenger()` to return the messenger rather than setting it?  then the instance variable can be private (with a public getter), and the base class' `initInstances()` method can set the instance variable to the value it gets back from `initBinaryMessenger()`.

Along with this change, maybe `initBinaryMessenger()` should be renamed `createBinaryMessenger()` or something similar?

Reply:
I was so dumb... Ya, that would totally work. Updated defaultBinaryMessenger to be a private field. PTAL.